docker/Dockerfile

```diff
@@ -67,7 +67,7 @@ ENV MINIFI_HOME $MINIFI_BASE_DIR/nifi-minifi-cpp-$MINIFI_VERSION
 RUN cd $MINIFI_BASE_DIR \
 	&& mkdir build \
 	&& cd build \
-	&& cmake .. \
+	&& cmake -DBUILD_SHARED_LIBS=OFF .. \
```

Comment:
what issues does this present?

Reply:
BUILD_SHARED_LIBS should only impact YAML. Is alpine failing with this?

Reply:
It is, looking back through the diff it appears I misread this and the fix may have been something of dumb luck.  The build is fine but we get something like the following without those disabled:
```
Error loading shared library libcivetweb.so.1.10.0: No such file or directory (needed by /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi)
Error loading shared library libcivetweb-cpp.so.1.10.0: No such file or directory (needed by /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi)
Error loading shared library libyaml-cpp.so.0.5: No such file or directory (needed by /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi)
Error loading shared library libtz.so: No such file or directory (needed by /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi)
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZTIN4YAML13BadConversionE: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZTIN4YAML13BadConversionE: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZTIN4YAML13BadConversionE: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZTI12CivetHandler: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZTI12CivetHandler: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler10handlePostEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler10handleHeadEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler10handleHeadEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler9handlePutEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler9handlePutEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler12handleDeleteEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler12handleDeleteEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler13handleOptionsEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler13handleOptionsEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler11handlePatchEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN12CivetHandler11handlePatchEP11CivetServerP13mg_connection: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN4YAML6detail9node_data3endEv: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN4YAML6detail9node_data8set_nullEv: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN4date12current_zoneEv: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN4YAML6detail6memory11create_nodeEv: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN4date11locate_zoneERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE: symbol not found
Error relocating /opt/minifi/nifi-minifi-cpp-0.6.0/bin/minifi: _ZN11CivetServerC1ESt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS6_EEPK14CivetCallbacksPKv: symbol not found
```

Reply:
Hmmm, that seems like something is hanging in the wrong place. I think there is a bigger issue. It shouldn't be using those symbols. Is this a blocker for you? Happy to merge if it fixes issue, but I am worried there is something more nefarious at work.

Reply:
I don't think there is a pressing need but would like ot get resolved before release.

Reply:
k thanks. I'll take a look and merge in the next day or so if I can't get anywhere with it.

Reply:
These should only be needed with ENABLE_PYTHON, so we could probably avoid this entirely with a separate conditional. Almost have a PR ready for that. Should alleviate this issue -- and make this PR unnecessary.

Reply:
sounds good.  happy to verify that PR and the docker build when available.  thank you

Reply:
https://github.com/apache/nifi-minifi-cpp/pull/419

Reply:
appveyor passed with it again -- though there was a spurious test failure. I've seen that test fail with port collisions -- so that's safe to ignore for the time being